### PR TITLE
make all imports of BuildEnv.props conditional 

### PR DESCRIPTION
**Pull request by smaillet-ms**
Fix for issue #15 
Made all imports of BuildEnv.props conditional as some of the project files are copied (and transformed) into the output location before being built, thus BuildEnv.props won't be in the path anywhere. This only occurs when doing an SDK build when the environment is already set up. BuildEnv.props isn't needed in that case, so not importing it is OK.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,5 +1,5 @@
 # Build results
-BuildOutput/
+/BuildOutput
```
Comment (mortezag) [inline .gitignore]: What is the reason of this change?

**Comment by mortezag** [comment]
:+1:

File: .gitignore (hunk 1 above)
Comment (smaillet-ms) [inline .gitignore]: VS integration was showing the BuildOutput as a non-ignored folder. So I used VS to set the ignore and this is the syntax it uses. (It's a bit more correct as the intent is to ignore the %SPOCLIENT%\BuildOutput folder as opposed to any folder under %SPOCLIENT named BuildOutput)
